Sources/StreamChat/Database/DTOs/ThreadReadDTO.swift

```diff
@@ -63,4 +87,61 @@ extension NSManagedObjectContext {
         dto.unreadMessagesCount = Int64(payload.unreadMessagesCount)
         return dto
     }
+
+    func loadThreadRead(parentMessageId: MessageId, userId: String) -> ThreadReadDTO? {
+        ThreadReadDTO.load(parentMessageId: parentMessageId, userId: userId, context: self)
+    }
+
+    func loadThreadReads(for userId: UserId) -> [ThreadReadDTO] {
+        ThreadReadDTO.load(userId: userId, context: self)
+    }
+
+    func markThreadAsRead(parentMessageId: MessageId, userId: UserId, at readAt: Date) {
+        if let read = loadThreadRead(parentMessageId: parentMessageId, userId: userId) {
+            read.lastReadAt = readAt.bridgeDate
+            read.unreadMessagesCount = 0
+        } else {
+            makeEmptyRead(parentMessageId: parentMessageId, userId: userId, readAt: readAt)
+        }
+    }
+
+    func markThreadAsUnread(
+        for parentMessageId: MessageId,
+        userId: UserId
+    ) {
+        let read = loadThreadRead(parentMessageId: parentMessageId, userId: userId)
+            ?? makeEmptyRead(parentMessageId: parentMessageId, userId: userId, readAt: nil)
+
+        // At the moment, the backend sets the value to 1
+        // Although ideally it should be equal to the replyCount?
```

Comment:
Since we only allow marking the whole thread, I think having 1 is also fine. If we allow also a particular message in the thread to be marked unread, then we would need something better.